dashboard/app/models/user.rb

```diff
@@ -1234,6 +1236,13 @@ def send_reset_password_instructions(email)
     return nil
   end
 
+  def reset_secrets
+    generate_secret_picture
+    generate_secret_words
+
+    self
```

Comment:
The return value of this method is unused and untested - I would've just cut these last two lines.  Is there a best practice here I'm missing?

Reply:
i would expect this method to either return the user (so you could use [builder-pattern-style](https://medium.com/@dljerome/design-patterns-in-ruby-builder-6c6d172e8de8) method chaining) or the newly-updated data (like `generate_secret_words`), but that may be subjective rather than a best practice

Reply:
@madelynkasula I don't quite get how this relates to the builder pattern, since it's returning self whereas builder AFAIK is a factory with stateful configuration, meaning it creates new objects. It is useful for chaining though, and I'm all for that if it's used, but like Brad said it doesn't appear to be and is not tested. 

If we want to keep the return for chaining, please add [documentation](https://yardoc.org/) with a `@return` and a unit test.

Reply:
This is a bit of an odd case for builder-pattern since (in theory) any newly-created user would have random secrets and it would be redundant to randomize them again.  I guess I'd lean toward returning the newly-updated data, since that's what `generate_secret_picture` and `generate_secret_words` are, in turn, doing.

Reply:
i think you're both right that it doesn't make sense to `return self` just for the purpose of method chaining if it isn't used. and, now that i think about it more, i hesitate to return the newly-updated data (as a hash of `secret_picture` and `secret_words`) because that wouldn't be used either.

would it make sense to just remove lines 1242 and 1243? (along with adding a test to make sure the two methods get called)